src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/GroupManager.java

```diff
@@ -252,11 +286,34 @@ public void addUserToGroup(final UserGroupDTO group, final RegisteredUserDTO use
 
         } else {
             // otherwise it is a noop.
+            // although we should force the user membership status to be active for the group.
+            this.setMembershipStatus(group, userToAdd, GroupMembershipStatus.ACTIVE);
             log.info(String.format("User (%s) is already a member of the group with id %s. Skipping.",
                     userToAdd.getId(), group.getId()));
         }
     }
 
+    /**
+     * Change users group membership status
+     *
+     * @param group
+     *            - that should be affected
+     * @param user
+     *            - user that should be affected.
+     * @param newStatus
+     *            - the new membership status
+     * @throws SegueDatabaseException
+     *             - If an error occurred while interacting with the database.
+     */
+    public void setMembershipStatus(final UserGroupDTO group, final RegisteredUserDTO user, GroupMembershipStatus newStatus)
+            throws SegueDatabaseException {
+        Validate.notNull(group);
+        Validate.notNull(user);
+        // we don't want people to delete user membership via this route as observers are not notified.
+        Validate.isTrue(!GroupMembershipStatus.DELETED.equals(newStatus), "Deletion of a group membership should not use this route.");
```

Comment:
One small suggestion: instead of Validate.isTrue(...) doing a run-time check, we could have two Enums, one inheriting the other, with the subclass also including delete. Then we could use types to enforce this at compile-time. Probably not worth the work now but an idea for the future.